src/kafka/consumer.go

```diff
@@ -134,15 +136,24 @@ func (k *kafka) Start(signals chan os.Signal, notifications chan<- Notification)
 	}
 }
 
-func (k *kafka) worker(consumer *cluster.Consumer, buffSize int, notifications chan<- Notification) {
+func batchDeadlineExceeded(lastReceivedMsg *time.Time, batchDeadline time.Duration) bool {
+	if lastReceivedMsg == nil {
+		return false
+	}
+
+	return time.Now().Sub(*lastReceivedMsg) > batchDeadline
+}
+
+func (k *kafka) worker(consumer *cluster.Consumer, buffSize int, batchDeadline time.Duration, notifications chan<- Notification) {
 	buf := make([]*sarama.ConsumerMessage, buffSize)
 	var decoded []*models.Record
+	var lastReceivedMsg *time.Time
 	idx := 0
 	for {
 		kafkaMsg := <-k.consumerCh
 		buf[idx] = kafkaMsg
 		idx++
-		for idx == buffSize {
+		if idx == buffSize || batchDeadlineExceeded(lastReceivedMsg, batchDeadline) {
```

Comment:
I guess the logic here is not as intended. Imagine this scenario when deadline is set to `1 minute`:
- We get a single message from kafka, it goes into `consumerCh`, we consume it but it does not hit the buffer size or the deadline
- Time passes without any message (say, 2 hours)
- The first message will only be flushed when another message arrives

I guess that's not the desired behavior